include/openPMD/Iteration.hpp

```diff
@@ -158,10 +163,59 @@ class Iteration : public Attributable
      * but not necessarily yet in the backend.
      * Will be propagated to the backend upon next flush.
      * Store the current status.
+     * Once an iteration has been closed, no further flushes shall be performed.
+     * If flushing a closed file, the old file may otherwise be overwritten.
      */
     std::shared_ptr< CloseStatus > m_closed =
         std::make_shared< CloseStatus >( CloseStatus::Open );
 
+    /**
+     * Whether a step is currently active for this iteration.
+     * Used for file-based iteration layout, see Series.hpp for
+     * group-based layout.
+     * Access via stepStatus() method to automatically select the correct
+     * one among both flags.
+     */
+    std::shared_ptr< StepStatus > m_stepStatus =
+        std::make_shared< StepStatus >( StepStatus::NoStep );
+
+    /**
+     * @brief Begin an IO step on the IO file (or file-like object)
+     *        containing this iteration. In case of group-based iteration
+     *        layout, this will be the complete Series.
+     *
+     * @return AdvanceStatus
+     */
+    AdvanceStatus
+    beginStep();
+
+    /**
+     * @brief End an IO step on the IO file (or file-like object)
+     *        containing this iteration. In case of group-based iteration
+     *        layout, this will be the complete Series.
+     *
+     * @return AdvanceStatus
+     */
+    void
+    endStep();
+
+    /*
+     * We cannot give the return type yet, since Iteration is still an
+     * incomplete type.
```

Comment:
I don't understand what `myIteration` is doing from the Doxygen string. Can you please clarify? What does this iterate over if `Iteration` is a single openPMD iteration?

It this maybe better implemented as an external function like `iter`/`next` or `std::begin()` specialization, etc.? Bit curious, looks like something that would belong to `Series`.

Reply:
Yeah, moved this one to `Series` now.